cpp_tests/feature/one_hot_encoder_test.cc

```diff
@@ -39,24 +38,17 @@ class OneHotEncoderTest : public testing::Test {
     mlir_context->appendDialectRegistry(registry);
   }
 
-  static void TearDownTestSuite() {
-    delete host_context;
-    host_context = nullptr;
-    delete mlir_context;
-    mlir_context = nullptr;
-  }
-
-  static tfrt::HostContext *host_context;
-  static MLIRContext *mlir_context;
+  static std::unique_ptr<tfrt::HostContext> host_context;
+  static std::unique_ptr<MLIRContext> mlir_context;
```

Comment:
What is the benefit of changing its type to `std::unique_ptr<...>`?

Reply:
The purpose is to keep all allocated memory chunks under control. In order to achieve this we have two options:

- Use `RCReference` and `LeakCheckAllocator`. In this case we still need to explicitly free the memory chunks allocated in JNA methods, but `LeakCheckAllocator` will check whether we have explicitly freed them.
- Use smart pointers like `std::unique_ptr`. In this case we don't free the memory chunks as these pointers can manage themselves wisely.

For `HostContext` I think the second option is better as we create `HostContext`s with `CreateHostContext`, whose return value has been `unique_ptr`.

If we still use raw pointers like `HostContext*`, it is true that we can check to make sure this pointer will always be freed in this PR. But I'm worried that as test files in Clink grow so that there are numerous  `new`/`delete` invocations in the project, it will be difficult for reviewers to judge whether a new PR has introduced unnecessary memory allocation behaviors, thus increasing memory leak risk and the workload to check it. It can be resolved if we prohibit any `new` in Clink.

In #8 we agreed to change it from `unique_ptr` to pointers because smart pointers might only be freed when all test cases finished, so the change could avoid having a lot of `HostContext` existing in memory. I think the purpose to avoid memory leak is more important than this one, so I changed it back to pointers in this PR.